apps/emqx_bridge/test/emqx_bridge_api_SUITE.erl

```diff
@@ -550,6 +551,89 @@ t_http_crud_apis(Config) ->
 
     {ok, 204, <<>>} = request(delete, uri(["bridges", BridgeID]), Config).
 
+t_auto_restart_interval(Config) ->
+    Port = ?config(port, Config),
+    %% assert we there's no bridges at first
+    {ok, 200, []} = request_json(get, uri(["bridges"]), Config),
+
+    meck:new(emqx_resource, [passthrough]),
+    meck:expect(emqx_resource, call_start, fun(_, _, _) -> {error, fake_error} end),
+
+    %% then we add a webhook bridge, using POST
+    %% POST /bridges/ will create a bridge
+    URL1 = ?URL(Port, "path1"),
+    Name = ?BRIDGE_NAME,
+    BridgeID = emqx_bridge_resource:bridge_id(?BRIDGE_TYPE_HTTP, Name),
+    BridgeParams = ?HTTP_BRIDGE(URL1, Name)#{
+        <<"resource_opts">> => #{<<"auto_restart_interval">> => "1s"}
+    },
+    ?check_trace(
+        begin
+            ?assertMatch(
+                {ok, 201, #{
+                    <<"type">> := ?BRIDGE_TYPE_HTTP,
+                    <<"name">> := Name,
+                    <<"enable">> := true,
+                    <<"status">> := _,
+                    <<"node_status">> := [_ | _],
+                    <<"url">> := URL1
+                }},
+                request_json(
+                    post,
+                    uri(["bridges"]),
+                    BridgeParams,
+                    Config
+                )
+            ),
+            ?block_until(#{?snk_kind := resource_disconnected_enter}),
+            ?block_until(#{?snk_kind := resource_auto_reconnect}, 1500)
+        end,
+        fun(Trace0) ->
+            Trace = ?of_kind(resource_auto_reconnect, Trace0),
+            ?assertMatch([#{}], Trace),
+            ok
+        end
+    ),
+    %% delete the bridge
+    {ok, 204, <<>>} = request(delete, uri(["bridges", BridgeID]), Config),
+    {ok, 200, []} = request_json(get, uri(["bridges"]), Config),
+
+    %% auto_retry_interval=infinity
+    BridgeParams1 = BridgeParams#{
+        <<"resource_opts">> => #{<<"auto_restart_interval">> => "infinity"}
+    },
+    ?check_trace(
+        begin
+            ?assertMatch(
+                {ok, 201, #{
+                    <<"type">> := ?BRIDGE_TYPE_HTTP,
+                    <<"name">> := Name,
+                    <<"enable">> := true,
+                    <<"status">> := _,
+                    <<"node_status">> := [_ | _],
+                    <<"url">> := URL1
+                }},
+                request_json(
+                    post,
+                    uri(["bridges"]),
+                    BridgeParams1,
+                    Config
+                )
+            ),
+            ?block_until(#{?snk_kind := resource_disconnected_enter}),
```

Comment:
I think you need to check the return val of `?block_until` since it could return `timeout`